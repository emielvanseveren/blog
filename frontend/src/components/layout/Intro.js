import React from 'react'
import styled from 'styled-components'
import Link from './Link'
import size from '../../constants/size'

import me from '../../images/me.jpg'

const Container = styled.aside`
  width: 100%;
  padding-left: 25px;
  display: flex;
  align-items: flex-start;
  margin-bottom: 125px;
  @media ${size.sm}{
    margin-bottom:50px;
    padding-left: 0;
    width: 95%;
    margin-left: auto;
    margin-right: auto;
  }
`
const CircleImg = styled.img`
  display: inline-block;
  width: 75px;
  height: 75px;
  border-radius: 50%;

`
const TextContainer = styled.div`
  margin-left: 25px;
  height: 100%;
  display: inline-flex;
  justify-content: center;
  flex-direction: column;
  h1{
    font-size: 1.2rem;
  }
  @media ${size.xs}{
    margin-left: 10px;
  }

`

export default function Intro(){
  return (
    <Container>
      <CircleImg alt="circle image Emiel Van Severen" src={me}/>
      <TextContainer>
        <h1>Emiel Van Severen</h1>
        <p>If you can´t explain it simply, you don´t understand it well enough.</p>
        <Link internal={false} to="https://github.com/emielvanseveren">github.com/emielvanseveren</Link>
      </TextContainer>
    </Container>
  )
}
