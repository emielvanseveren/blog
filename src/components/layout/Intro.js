import React from 'react'
import styled from 'styled-components'
import Link from './Link'

import me from '../../images/me.jpg'

const Container = styled.aside`
  width: 100%;
  padding-left: 25px;
  display: flex;
  align-items: top;
`
const CircleImg = styled.img`
  display: inline-block;
  width: 75px;
  height: 75px;
  border-radius: 50%;
`
const TextContainer = styled.div`
  margin-left: 25px;
  height: 100%;
  display: inline-flex;
  justify-content: center;
  flex-direction: column;
  h1{
    font-size: 1.2rem;
  }
`

export default function Intro(){
  return (
    <Container>
      <CircleImg alt="circle image Emiel Van Severen" src={me}/>
      <TextContainer>
        <h1>Emiel Van Severen</h1>
        <p>If you can´t explain it simply, you don´t understand it well enough.</p>
        <Link internal={false} to="https://github.com/emielvanseveren">github.com/emielvanseveren</Link>
      </TextContainer>
    </Container>
  )
}
