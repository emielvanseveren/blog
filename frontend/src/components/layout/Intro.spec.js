import React from 'react'
import Intro from './Intro'

describe('<Intro/>', () => {
  it('Snapshot', () => {
    expect(mountWithRouterAndTheme(
      <Intro/>
    )).toMatchSnapshot()
  })
})
