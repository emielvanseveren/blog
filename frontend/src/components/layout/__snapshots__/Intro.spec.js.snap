// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Intro/> Snapshot 1`] = `
<Intro>
  <Intro__Container>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Intro__Container-n2wl2x-0",
            "isStatic": false,
            "lastClassName": "gCFbfp",
            "rules": Array [
              "width:100%;padding-left:25px;display:flex;align-items:flex-start;margin-bottom:125px;@media ",
              "(max-width: 768px)",
              "{margin-bottom:50px;padding-left:0;width:95%;margin-left:auto;margin-right:auto;}",
            ],
          },
          "displayName": "Intro__Container",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Intro__Container-n2wl2x-0",
          "target": "aside",
          "toString": [Function],
          "usesTheme": false,
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <aside
        className="Intro__Container-n2wl2x-0 gCFbfp"
      >
        <Intro__CircleImg
          alt="circle image Emiel Van Severen"
          src=""
        >
          <StyledComponent
            alt="circle image Emiel Van Severen"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "Intro__CircleImg-n2wl2x-1",
                  "isStatic": false,
                  "lastClassName": "huXkvm",
                  "rules": Array [
                    "display:inline-block;width:75px;height:75px;border-radius:50%;",
                  ],
                },
                "displayName": "Intro__CircleImg",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "Intro__CircleImg-n2wl2x-1",
                "target": "img",
                "toString": [Function],
                "usesTheme": false,
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            src=""
          >
            <img
              alt="circle image Emiel Van Severen"
              className="Intro__CircleImg-n2wl2x-1 huXkvm"
              src=""
            />
          </StyledComponent>
        </Intro__CircleImg>
        <Intro__TextContainer>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "Intro__TextContainer-n2wl2x-2",
                  "isStatic": false,
                  "lastClassName": "jhaMeW",
                  "rules": Array [
                    "margin-left:25px;height:100%;display:inline-flex;justify-content:center;flex-direction:column;h1{font-size:1.2rem;}@media ",
                    "(max-width: 576px)",
                    "{margin-left:10px;}",
                  ],
                },
                "displayName": "Intro__TextContainer",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "Intro__TextContainer-n2wl2x-2",
                "target": "div",
                "toString": [Function],
                "usesTheme": false,
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="Intro__TextContainer-n2wl2x-2 jhaMeW"
            >
              <h1>
                Emiel Van Severen
              </h1>
              <p>
                If you can´t explain it simply, you don´t understand it well enough.
              </p>
              <Link
                internal={false}
                to="https://github.com/emielvanseveren"
              >
                <Link__ExternalLink
                  href="https://github.com/emielvanseveren"
                  target="_blank"
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "Link__ExternalLink-y16ojc-1",
                          "isStatic": false,
                          "lastClassName": "iJsnQe",
                          "rules": Array [
                            "text-decoration-line:underline;text-decoration-color:",
                            [Function],
                            ";color:",
                            [Function],
                            ";font-size:110%;@media ",
                            "(max-width: 768px)",
                            "{font-size:90%;}",
                          ],
                        },
                        "displayName": "Link__ExternalLink",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "Link__ExternalLink-y16ojc-1",
                        "target": "a",
                        "toString": [Function],
                        "usesTheme": false,
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    href="https://github.com/emielvanseveren"
                    target="_blank"
                  >
                    <a
                      className="Link__ExternalLink-y16ojc-1 iJsnQe"
                      href="https://github.com/emielvanseveren"
                      target="_blank"
                    >
                      github.com/emielvanseveren
                    </a>
                  </StyledComponent>
                </Link__ExternalLink>
              </Link>
            </div>
          </StyledComponent>
        </Intro__TextContainer>
      </aside>
    </StyledComponent>
  </Intro__Container>
</Intro>
`;
